crates/ggml/src/context.rs

```diff
@@ -473,6 +473,20 @@ impl Context {
         let tensor = unsafe { sys::ggml_gelu(self.ptr.as_ptr(), a.ptr.as_ptr()) };
         self.new_tensor_raw(tensor)
     }
+
+    /// flash attention.
+    pub fn op_flash_attn(&self, q: &Tensor, k: &Tensor, v: &Tensor, masked: bool) -> Tensor {
+        let tensor = unsafe {
+            sys::ggml_flash_attn(
+                self.ptr.as_ptr(),
+                q.ptr.as_ptr(),
+                k.ptr.as_ptr(),
+                v.ptr.as_ptr(),
+                masked,
+            )
+        };
```

Comment:
I don't know if we should use ggmls flash-attention. The ggml version of the main branch is quite old (~3 weeks) mybe this could lead to errors 🤔

Reply:
I noticed that the upstream already supports broadcast `mut_mat`, perhaps we can use it to replace `repeat`.

Reply:
We could wait for the CUDA PR to get merged as it is already based on the newest `ggml` version, then we could try to use the new `mut_mat`. What do you think?

Reply:
Sounds good.